engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/utils/BigIntegerOps.java

```diff
@@ -0,0 +1,85 @@
+package org.enso.interpreter.node.expression.builtin.number.utils;
+
+import com.oracle.truffle.api.CompilerDirectives;
+
+import java.math.BigInteger;
+
+/**
+ * Re-exposes big-integer operations behind a truffle boundary.
```

Comment:
I'm still not sure why we don't want to inline these.

Reply:
Because these are operations containing a lot of code and at the same time the probability of any improvement coming from inlining them is almost zero.